### PR TITLE
Fix issue #7018

### PR DESCRIPTION
**Pull request by jlainezs**
Pull Request for Issue #7018 .

### Summary of Changes
Removed call to parent::publish in MenusMoldeItem::publish to avoid use the JTableNested::publish which unpublishes the selected item and all its children.

### Testing Instructions
Open the menu items page and click on any menu item. If it has children only the item will change its state. Select several items and click the publish or unpublish buttons on the top par. Only selected items will change its state.

### Documentation Changes Required
Not known.

### CODE REVIEW

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on d2743095d628c3881b8c112f8371a8c78336f2c9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13242">issues.joomla.org/tracker/joomla-cms/13242</a>.</sub>

**Comment by csthomas** [comment]
IMHO `publish` does not have to trigger publish on children items,
but `unpublish` should trigger unpublish on children items that are (published or archived) - for security

**Comment by mbabker** [comment]
> IMHO publish does not have to trigger publish on children items,
> but unpublish should trigger unpublish on children items that are (published or archived) - for security

Correct.  In a tree hierarchy, you should not have children of an unpublished item in an accessible state.  So an unpublish action should ensure children correctly unpublish, but this same cascade doesn't immediately apply to a publish action.

File: administrator/manifests/packages/pkg_weblinks.xml [hunk 1]
```diff
@@ -1,30 +0,0 @@
-<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (Sieger66) [inline administrator/manifests/packages/pkg_weblinks.xml]: Why do you delete this

Comment (jlainezs) [inline administrator/manifests/packages/pkg_weblinks.xml]: didn't noticed about this.

**Comment by Sieger66** [comment]
Then correct your PR

**Comment by jlainezs** [comment]
Oks!

File: administrator/components/com_menus/models/item.php [hunk 2]
```diff
@@ -1628,7 +1628,30 @@ public function publish(&$pks, $value = 1)
...
-		return parent::publish($pks, $value);
+		$result = false;
+
+		try
+		{
+			$pks = ArrayHelper::toInteger($pks);
+
+			if (!empty($pks))
+			{
+				$dbo = $table->getDbo();
```
Comment (yvesh) [inline administrator/components/com_menus/models/item.php]: Call it $db please, we are all so used to it :-)

**Comment by jlainezs** [comment]
No problem, it is better to follow the existing conventions :)

**Comment by jlainezs** [comment]
Oks... file restored and corrected the variable name as @yvesh told. If there's anything more, please, let me know.

**Comment by Sieger66** [comment]
I agree with csthomas and mbabker :

> unpublish should trigger unpublish on children items that are (published or archived) - for security

this not do your PR 

Please complete your PR with this.

**Comment by jlainezs** [comment]
@Sieger66, @csthomas or @mbabker  When you say 

> unpublish should trigger unpublish on children items that are (published or archived) - for security

do you mean that on publish it should skip unpublished children?
Thanks!

**Comment by Sieger66** [comment]
It mean when you unpublish a parent it is need also to unpublish all his children items.
Before your PR this is also.

You PR is correct when you publish a parent is need nothing to do with the children items.

**Comment by jlainezs** [comment]
Oks... now is clear :)

**Comment by csthomas** [comment]
I have prepared a code, which is related to the desired behaviour but did it more global.
Maybe you want to look at #14439 It is for all nested tables as menu and categories.

**Comment by jlainezs** [comment]
oks... then I only need to undo changes and left it as it was before any 
change.
For me it is better than solve the issue in the menu, so in a few hours 
I will revert changes I did to the menu parent. Is this right?

**Comment by csthomas** [comment]
Now I do not know if my changes will be accepted because it changes a lot of things. 
You can use the same rules (for recursion) as in my PR (but only for menu) and then it should be OK.

My applied rules:
* If publish item then do nothing on children
* If unpublish item then unpublish all published and archived children
* If archive item then archive published children too
* If trash item then trash all children

**Comment by jlainezs** [comment]
oks.. I've seen the comments a few minutes ago

**Comment by Sieger66** [comment]
I have tested this item :white_check_mark: successfully on a9a69526567f831bc09dc61f9501053be01fd7c3<br /><br />But if you unpublish an parent all his children items with status trashed also become status unbublish.
Is no need to change status from trashed to unpublish if we unpublish an parent menu item.

But the solution its better as before and o.k. for me. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13242">issues.joomla.org/tracker/joomla-cms/13242</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC as there are 2 successfully Tests?

**Comment by brianteeman** [comment]
If something is trashed.it shouldn't come back

**Comment by Sieger66** [comment]
But we have only one successfully tests.
Your test is with a old stand in this PR and not with the last stand.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13242">issues.joomla.org/tracker/joomla-cms/13242</a>.</sub>

**Comment by jlainezs** [comment]
as it is coded check to no unpublish trashed ítems as well as archived
items seems easy. So pls., tell me what to do.

El dia 18/03/2017 19.27, "Sieger66" <notifications@github.com> va escriure:

**Comment by infograf768** [comment]
What I tested:
Publishing Parent Item has no effect on children => OK
Unpublishing Parent Item does Unpublish children => OK, including Trashed Items=> Wrong
So this has to be solved.

Trashing Parent item does also trash children => Not sure about that one

We do not Archive menu items as far as I know.

**Comment by jlainezs** [comment]
Thanks for the feedback. I'll do the PR along the week.

**Comment by csthomas** [comment]
@jlainezs After #14439 has been merged you should check again if this PR is needed.

**Comment by jlainezs** [comment]
After #14439 <https://github.com/joomla/joomla-cms/pull/14439> merge I 
think this PR is not needed.

**Comment by zero-24** [comment]
so we should close here @jlainezs ?

**Comment by jlainezs** [comment]
Yes, please.

**Comment by zero-24** [comment]
closing as requested.

**Pull request closed without merging**